File: 2023-2024/sourcecode/mod2/lab9/RepositoryBasicsJava/app/src/main/java/com/example/repositorybasicsjava/ui/UserViewHolder.java
package com.example.repositorybasicsjava.ui;

import androidx.recyclerview.widget.RecyclerView;

import com.example.repositorybasicsjava.databinding.RvItemBinding;
import com.example.repositorybasicsjava.model.User;

public class UserViewHolder extends RecyclerView.ViewHolder {
    private RvItemBinding binding;

    public UserViewHolder(RvItemBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(User item) {
        String fullName = item.getFirstName() + " " + item.getLastName();
        binding.userTextView.setText(fullName);
    }
}

File: 2023-2024/sourcecode/mod3/lab9/RepositoryBasicsKotlin/app/src/main/java/com/example/repositorybasicskotlin/ui/UserViewHolder.kt
package com.example.repositorybasicskotlin.ui

import androidx.recyclerview.widget.RecyclerView
import com.example.repositorybasicskotlin.databinding.RvItemBinding
import com.example.repositorybasicskotlin.model.User

class UserViewHolder(private val binding: RvItemBinding) : RecyclerView.ViewHolder(binding.root) {
    fun bind(item: User) {
        val fullName = "${item.firstName} ${item.lastName}"
        binding.userTextView.text = fullName
    }
}